src/CognitoIdentity/CognitoIdentityClient.php

```diff
@@ -43,4 +44,7 @@
  * @method \Aws\Result updateIdentityPool(array $args = [])
  * @method \GuzzleHttp\Promise\Promise updateIdentityPoolAsync(array $args = [])
  */
-class CognitoIdentityClient extends AwsClient {}
+class CognitoIdentityClient extends AwsClient
+{
+    use CredentialCreatingClientTrait;
```

Comment:
It seems weird to duplicate the `createCredentialsMethod` on multiple clients when they do exactly the same thing and don't really create credentials specific to the client. I wonder if we should just add a factory/creational function to Credentials that creates them from an array, mark the Sts method `createCredentials` deprecated, and make that simply proxy to the factory method.